chromadb/auth/__init__.py

```diff
@@ -1,449 +1,238 @@
-        ...
-
-    @abstractmethod
-    def get_user_databases(self) -> Optional[List[str]]:
-        ...
-
-    @abstractmethod
-    def get_user_attributes(self) -> Optional[Dict[str, Any]]:
-        ...
-
-
-class SimpleUserIdentity(UserIdentity):
-    def __init__(
-        self,
-        user_id: str,
-        tenant: Optional[str] = None,
-        databases: Optional[List[str]] = None,
-        attributes: Optional[Dict[str, Any]] = None,
-    ) -> None:
-        self._user_id = user_id
-        self._tenant = tenant
-        self._attributes = attributes
-        self._databases = databases
-
-    @override
-    def get_user_id(self) -> str:
-        return self._user_id
-
-    @override
-    def get_user_tenant(self) -> Optional[str]:
-        return self._tenant if self._tenant else DEFAULT_TENANT
-
-    @override
-    def get_user_databases(self) -> Optional[List[str]]:
-        return self._databases
-
-    @override
-    def get_user_attributes(self) -> Optional[Dict[str, Any]]:
-        return self._attributes
-
-
-class ClientAuthResponse(EnforceOverrides, ABC):
-    @abstractmethod
-    def get_auth_info_type(self) -> AuthInfoType:
-        ...
-
-    @abstractmethod
-    def get_auth_info(
-        self,
-    ) -> Union[Tuple[str, SecretStr], List[Tuple[str, SecretStr]]]:
-        ...
+ClientAuthHeaders = Dict[str, SecretStr]
 
 
 class ClientAuthProvider(Component):
+    """
+    ClientAuthProvider is responsible for providing authentication headers for
+    client requests. Client implementations (in our case, just the FastAPI
+    client) must inject these headers into their requests.
+    """
     def __init__(self, system: System) -> None:
         super().__init__(system)
 
     @abstractmethod
-    def authenticate(self) -> ClientAuthResponse:
+    def authenticate(self) -> ClientAuthHeaders:
         pass
 
 
-class ClientAuthConfigurationProvider(Component):
-    def __init__(self, system: System) -> None:
-        super().__init__(system)
-
-    @abstractmethod
-    def get_configuration(self) -> Optional[T]:
-        pass
-
+@dataclass
+class UserIdentity:
+    """
+    UserIdentity represents the identity of a user. In general, not all fields
+    will be populated, and the fields that are populated will depend on the
+    authentication provider.
 
-class ClientAuthCredentialsProvider(Component, Generic[T]):
-    def __init__(self, system: System) -> None:
-        super().__init__(system)
+    The idea is that the AuthenticationProvider is responsible for populating
+    _all_ information known about the user, and the AuthorizationProvider is
+    responsible for making decisions based on that information.
+    """
+    user_id: str
+    tenant: Optional[str] = None
+    databases: Optional[List[str]] = None
+    # This can be used for any additional auth context which needs to be
+    # propagated from the authentication provider to the authorization
+    # provider.
+    attributes: Optional[Dict[str, Any]] = None
```

Comment:
nit: should this be Any? can we be narrow and open it up as needed

What is this used for today?

Reply:
It's not used for anything -- the intent is to give anyone who implements authn and authz themselves a place to put any other data relevant to their use-case. For this reason I think it needs to be `Any` but I'd be happy to change it.